File: SUSTech/CS203/2020Fall/Stone Game/std.cpp
#include <iostream>
using namespace std;

const int maxn = 1e5 + 5;

int T, n;
int a[maxn];

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    cin >> T;
    while (T--) {
        cin >> n;
        for (int i = 1; i <= n; ++i) cin >> a[i];
        int XOR = 0, ans = 0;
        for (int i = 1; i <= n; ++i) XOR ^= a[i];
        if (XOR) {
            for (int i = 1; i <= n; ++i) {
                if (a[i] >= (XOR ^ a[i])) ++ans;
            }
        }
        cout << ans << '\n';
    }

    return 0;
}

File: SUSTech/CS203/2020Fall/Stone Game/std.java
import java.util.*;

public class Main {
    public static void main (String[] args) {
        Scanner in = new Scanner(System.in);
        int T = in.nextInt();
        while (T-- > 0) {
            int n = in.nextInt();
            int[] a = new int[n + 1];
            for (int i = 1; i <= n; ++i) a[i] = in.nextInt();
            int XOR = 0;
            int ans = 0;
            for (int i = 1; i <= n; ++i) XOR ^= a[i];
            if (XOR != 0) {
                for (int i = 1; i <= n; ++i) if (a[i] >= (XOR ^ a[i])) ++ans;
            }
            System.out.println(ans);
        }
    }
}